### DISCUSSION TITLE
mailer_dsn with TLS and port 587

### DISCUSSION THREAD

**Discussion by beedaddy** [wallabag/wallabag#7241] [Q&A] [answered]
I just installed wallabag 2.6.8 and having some problems finding a working mailer_dsn configuration. Maybe someone could provide me an example mailer_dsn string for TLS and port 587? In my case, the mail user name contains a '@', which might be a problem here. I tried both `smtp://noreply@mydomain.com:mypassword123@mail.mydomain.com:587` as well as `smtp://noreply%40mydomain.com:mypassword123@mail.mydomain.com:587` in order to escape the '@' in the username.

The result is a timeout and: Expected response code "235" but got code "435".

**Comment by HolgerAusB** [comment]
https://doc.wallabag.org/en/admin/mailer advises to double-escape special characters.

Try:
`smtp://noreply%%40mydomain.com:mypassword123@mail.mydomain.com:587`

But I am not shure, if you can use this mode with TLS encryption

**Reply by beedaddy** [reply]
Thanks for the hint, but still no success. The complete error message is:
```
500: Internal Server Error

Failed to authenticate on SMTP server with username "noreply" using the following 
authenticators: "LOGIN", "PLAIN". Authenticator "LOGIN" returned "Expected response
code "235" but got code "435", with message "435 4.7.8 Error: authentication failed:
UGFzc3dvcmQ6".". Authenticator "PLAIN" returned "Connection to "mail.mydomain.com:587"
 timed out.".
```
"UGFzc3dvcmQ6" indicates that the server detected a wrong password (google search). What makes me wonder is that it's still talking about the username "noreply" and not "noreply@mydomain.com"...

**Comment by j0k3r** [comment]
Could you share the full line of the `mailer_dns` from the parameters.yml? Include mailer_dns?

**Reply by beedaddy** [reply]
Currently, it looks like this:
```yml
mailer_dsn: 'smtp://noreply%%40mydomain.com:myAlphaNumericPw@mail.mydomain.com:587'
```
(Of course, mydomain.com and the password (containing no special characters) are different.)

I also tried to append `/?encryption=tls`.

**Reply by j0k3r** [reply]
I'm out of idea.
Regarding the official doc for the mailer (ie: https://symfony.com/doc/4.x/mailer.html), this should work:

```yaml
mailer_dsn: 'smtp://noreply%40mydomain.com:myAlphaNumericPw@mail.mydomain.com:587'
```

**Reply by beedaddy** [reply]
Yes, but wait... Even if I enter nonsense, I get exactly this error message... 🤔

**Reply by beedaddy** [reply, marked as answer]
Yes, it was my fault. 
(me not clearing the cache 🙄)
